Support user data, init script and more!

RoryHunter opened the pull request:
Support for configuring:
* Droplet user data
* Jenkins init script

Add a slave configure-entries.jelly to silence an exception when viewing the node's configuring. It's not great but it's better than nothing.

Check for an SSH connection less aggressively, and wait for the droplet to enter the ACTIVE state before doing so.

Increase default idle time to 60 minutes, since that's the minimum you'll pay for anyway.

anpieber commented on src/main/resources/com/dubture/jenkins/digitalocean/SlaveTemplate/config.jelly:
```diff
@@ -47,7 +47,15 @@
 
         <f:entry title="Idle termination time" field="idleTerminationInMinutes">
-            <f:textbox default="10" />
+            <f:textbox default="60" />
```
I'm not so happy with this change. The problem is that you pay for every started hour. If the build took 10 minutes and you're waiting for 60 minutes you already pay two hours. IMHO the only correct solution would be to completely change this feature to only a radio button: 

[ ] keep alive
[ ] shutdown at next full hour

Whilte the first one would never shutdown the container, the second one will keep the container alive till the next full hour is reached and the scheduling algorithm would almost take the container with the longest remaining time to live (but those two things could be two different features). Till we've something like this in place I think the 10 minutes are the saner default choice.

anpieber commented on src/main/java/com/dubture/jenkins/digitalocean/ComputerLauncher.java:
```diff
@@ -189,29 +189,45 @@ private int bootstrap(Connection bootstrapConn, Computer computer, PrintStream l
+
+                    case NEW:
+                        logger.println("Droplet in NEW state, waiting for it to become ACTIVE");
+                        throw new IOException("sleep");
```
tbh I dont like the approach of handling a regular Situation by an exception. sleep cont. should work fine here too, wouldn't it?

anpieber commented:
Thank you so much for your work btw! I really appreciate it.

RoryHunter commented:
I've reverted the timeout as you suggest - I'll see about a more clever retention strategy if I get a chance. I've also refactored the connection code quite a lot, I think it's much improved.

anpieber commented:
Thank you very much again! I'll cut a release right away.

chriswalters commented:
Prior to the timeout changes, my droplets were launching and remaining connected with no problem. With 0.6 I'm consistently getting a “Unexpected termination of channel” error. I tried setting the "Idle termination time" as high as 1000 (as well as empty and 0) and continued to get the error. I switched back to 0.5 and the error went away.

RoryHunter commented:
I saw this too. I'll push a revert ASAP. 

-- 
Rory Hunter

> On 11 Aug 2015, at 03:13, Chris Walters <notifications@github.com> wrote:
> 
> Prior to the timeout changes, my droplets were launching and remaining connected with no problem. With 0.6 I'm consistently getting a “Unexpected termination of channel” error. I tried setting the "Idle termination time" as high as 1000 (as well as empty and 0) and continued to get the error. I switched back to 0.5 and the error went away.
> 
> —
> Reply to this email directly or view it on GitHub.
>

RoryHunter commented:
PR  #14.

anpieber commented:
Released
